lib/cf/broker/src/routes/create-service.js

```diff
@@ -1,238 +1,102 @@
-  let ratingPlan;
-
-  if(isEmpty(plans))
-    ratingPlan = sampleRatingPlan(planId);
-  else {
-    const metrics = plans[0].plan.metrics;
-    ratingPlan = sampleRatingPlan(planId, config.buildRatingPlanMetrics(metrics));
-  }
 
-  createPlan('rating', ratingPlan, planId, (err, planId) => cb(err, plans, planId));
+const getSinglePlan = (serviceConfig) => {
+  const plans = serviceConfig ? serviceConfig.plans : undefined;
+  return plans ? plans[0] : undefined;
 };
 
-const createMeteringPlanMapping = (instanceId, planId, cb) => createPlanMapping('metering', planId, instanceId, cb);
+const getRequestParameters = (request) => ({
+  instance_id: request.params.instance_id,
+  service_id: request.body.service_id,
+  serviceConfig: request.body.parameters,
+  organization_guid: request.body.context && request.body.context.organization_guid,
+  space_guid: request.body.context && request.body.context.space_guid
+});
 
-const createPricingPlanMapping = (instanceId, planId, cb) => createPlanMapping('pricing', planId, instanceId, cb);
+/* eslint-disable complexity */
+const createServiceHandler = (clientsFactory, config, planBuilder) =>
+  async (request, response) => {
+    const requestParameters = getRequestParameters(request);
 
-const createRatingPlanMapping = (instanceId, planId, cb) => createPlanMapping('rating', planId, instanceId, cb);
+    debug('Creating service instance with parameters %o', requestParameters);
 
-const createPlansTask = (plans, planId, cb) =>
-  async.waterfall([
-    async.apply(createMeteringPlan, plans, planId),
-    createPricingPlan,
-    createRatingPlan], (err) => {
-    if (err) {
-      edebug('Error during plans creation %o', err);
-      cb(err);
+    if(requestParameters.serviceConfig && !config.isServiceConfigValid(requestParameters.serviceConfig)) {
+      edebug('Service configuration is not valid, %o', requestParameters.serviceConfig);
+      response.status(httpStatus.BAD_REQUEST).send({ description: 'Invalid service configuration.' });
       return;
     }
 
-    cb();
-  });
-
-const createMappingsTask = (instanceId, planId, cb) =>
-  async.parallel([
-    async.apply(createMeteringPlanMapping, instanceId, planId),
-    async.apply(createPricingPlanMapping, instanceId, planId),
-    async.apply(createRatingPlanMapping, instanceId, planId)
-  ], (err) => {
-    if (err) {
-      edebug('Error during plan mappings creation %o', err);
-      cb(err);
+    const planId = config.generatePlanId(requestParameters.instance_id, requestParameters.instance_id);
+    const userProvidedPlan = getSinglePlan(requestParameters.serviceConfig);
+
+    try {
+      if (userProvidedPlan && userProvidedPlan.resource_provider) {
+        if (!config.getMappingApi()) {
+          edebug('Cannot create service mapping. No url defined.');
+          response.status(httpStatus.BAD_REQUEST).send({
+            description: 'Provided plan is not valid. Resource provider mapping is not supported.'
```

Comment:
I think `Invalid plan: resource provider mapping is not supported.` fits better here.

We can ask users to remove resourceProvider **or** contact support/Abacus ops to find out why this is missing, but we have no way to know upfront if mapping API will ever be implemented/supported on a particular landscape/installation.